code/controllers/subsystem/overlays.dm

```diff
@@ -83,18 +87,26 @@ SUBSYSTEM_DEF(overlays)
 	if (!cached_icon) //not using the macro to save an associated lookup
 		cached_icon = list()
 		icon_states_cache[icon] = cached_icon
+		if (length(icon_states_cache) > MAX_ICON_CACHE_AMOUNT)
+			icon_states_cache.Cut(1, MAX_ICON_CACHE_AMOUNT*0.1)
```

Comment:
This `0.1` number is used twice, it should be moved out to a define--`ICON_CACHE_CLEANUP_THRESHOLD`, or something.